sdk/src/main/scala/com/horizen/SidechainBackup.scala

```diff
@@ -0,0 +1,66 @@
+package com.horizen
+
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import com.horizen.backup.BoxIterator
+import com.horizen.box.BoxSerializer
+import com.horizen.companion.SidechainBoxesCompanion
+import com.horizen.storage._
+import com.horizen.utils.{ByteArrayWrapper, BytesUtils}
+import org.apache.commons.io.FileUtils
+import scorex.util.ScorexLogging
+
+import java.io._
+import java.lang.{Byte => JByte}
+import java.util.{HashMap => JHashMap}
+import scala.util.{Failure, Success}
+
+class SidechainBackup @Inject()
+  (@Named("CustomBoxSerializers") val customBoxSerializers: JHashMap[JByte, BoxSerializer[SidechainTypes#SCB]],
+   @Named("StateStorage") val stateStorage: Storage,
+   @Named("BackupStorage") val backUpStorage: Storage,
+   @Named("BackUpper") val backUpper : BoxBackupInterface
+  ) extends ScorexLogging
+  {
+    protected val sidechainBoxesCompanion: SidechainBoxesCompanion =  SidechainBoxesCompanion(customBoxSerializers)
+    protected val sidechainStateStorage = new SidechainStateStorage(
+      stateStorage,
+      sidechainBoxesCompanion)
+    protected val backupStorage = new BackupStorage(backUpStorage, sidechainBoxesCompanion)
+
+
+    def createBackup(sidechainBlockIdToRollback: String, copyStateStorage: Boolean, dataDirPath: String): Unit = {
+      if (copyStateStorage) {
+        val stateStorage: File = new File(dataDirPath+ "/state")
+        val stateStorageBackup: File = new File(dataDirPath+ "/state_backup")
+
+        try {
+          FileUtils.copyDirectory(stateStorage, stateStorageBackup)
+        } catch {
+          case t: Throwable =>
+            log.error("Error during the copy of the StateStorage: ",t.getMessage)
+            throw new RuntimeException("Error during the copy of the StateStorage: "+t.getMessage)
+        }
+      }
+
+      sidechainStateStorage.rollback(new ByteArrayWrapper(BytesUtils.fromHexString(sidechainBlockIdToRollback))) match {
```

Comment:
Good point, since the copy is optional maybe have sense to rollback the original one, but we can also add a check that if we have the copy we rollback that one instead of the original.